src/main/java/com/exasol/releaserobot/github/GithubAPIAdapter.java

```diff
@@ -81,14 +81,21 @@ public void sendGitHubRequest(final URI uri, final String json) throws GitHubExc
                 .build();
         final HttpClient build = HttpClient.newBuilder().proxy(ProxySelector.getDefault()).build();
         try {
-            build.send(request, HttpResponse.BodyHandlers.ofString());
+            final HttpResponse<String> response = build.send(request, HttpResponse.BodyHandlers.ofString());
+            validateResponse(response);
         } catch (final IOException | InterruptedException exception) {
             Thread.currentThread().interrupt();
             throw new GitHubException("F-RR-GH-2: Exception happened during sending an HTTP request to the GitHub.",
                     exception);
         }
     }
 
+    private void validateResponse(final HttpResponse<String> response) throws GitHubException {
+        if (response.statusCode() < 200 || response.statusCode() >= 300) {
```

Comment:
isn't enough to check if the status is not 200?

`if (HttpURLConnection.HTTP_OK != response.statusCode()) { `

Reply:
GitHub sends me a 204 response, not 200. Maybe it's safer to have a range?

Reply:
According to https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_success , Http successful codes go from 200 to 208, so you could use that range.

Reply:
Actually, by reading the doc, it looks like Github will always send a 204 https://docs.github.com/en/free-pro-team@latest/rest/reference/repos

So better check only for that